src/plugins/intel_gpu/tests/test_utils/test_utils.h

```diff
@@ -571,6 +572,44 @@ T div_up(const T a, const U b) {
 
 double default_tolerance(data_types dt);
 
+class membuf : public std::streambuf
+{
+public:
+    membuf() : _pos(0) { }
+
+protected:
+    virtual int_type overflow (int_type c) {
+        _buf.emplace_back(c);
+        return c;
+    }
+
+    virtual int_type uflow() {
+        return (_pos < _buf.size()) ? _buf[_pos++] : EOF;
+    }
+
+private:
+    std::vector<int_type> _buf;
+    size_t _pos;
+};
+
+inline bool is_caching_test() {
+    std::string curr_test_suite_name(::testing::UnitTest::GetInstance()->current_test_suite()->name());
```

Comment:
That approach looks really fragile. It's better to have caching flag as arg of parameterized tests or just create single unit test dedicated for serialization flow checks.

Reply:
Considering the case of sharing an already created stream, the stream is received as a separate argument.

Reply:
I removed the function and added an arg for caching flag. Thank you.